falcon/inspect.py

```diff
@@ -753,7 +777,17 @@ def visit_app(self, app: AppInfo) -> str:
 # ------------------------------------------------------------------------
 
 
-def _get_source_info(obj, default='[unknown file]'):
+def _get_source_info(
+    obj: ModuleType
```

Comment:
union since iirc 3.7 and lower don't like this

Reply:
also these are internal function, `Any` there is most likely enough

same in the other functions that are similarly typed

Reply:
Any instead of all those types?

Reply:
Used Union instead of `|`

Reply:
> Any instead of all those types?

yes, we don't really gain anything from typing it as the union of a lot of things. We just pass to the inspect library. Also these are internal methods (as they are underscored), so they aren't used by the users

Reply:
I've discovered that when we do `from __future__ import annotations` we can use pipes

Reply:
Changed to Any.